ambry-store/src/main/java/com/github/ambry/store/PersistentIndex.java

```diff
@@ -511,6 +516,32 @@ List<LogSegment> getLogSegments() {
     return result;
   }
 
+  /**
+   * Update the partial log segment info including the partialLogSegmentCount and wastedLogSegmentSpace
+   */
+  void updatePartialLogSegmentInfo() {
+    int partialCount = 0;
+    long wastedSpace = 0;
+    LogSegment segment = log.getFirstSegment();
+    LogSegmentName journalName = null;
+    if (journal.getCurrentNumberOfEntries() != 0) {
+      journalName = journal.getFirstOffset().getName();
+    }
+    while (segment != null) {
+      // exclude journal
+      if (journalName != null && segment.getName().equals(journalName)) {
+        break;
+      }
+      if (segment.getEndOffset() < segment.getCapacityInBytes() * 0.8) {
```

Comment:
My understanding so far is that  `partialCount` is supposed to give us some idea around which log segments were good candidates for compaction but maybe were not compacted by the current scheme? If yes, then maybe `0.8` might not be a good threshold. Its totally fine if something that is 0.79 full and is not picked up for compaction. We should worry about something that is 0.3 full and is not picked up. So probably we should lower this threshold to something like `0.5` or `0.6`

Reply:
No, it's not for picking compaction candidates. We use it to monitor the problem caused by stats based compaction which leave a lot of unused space. 
This is not the valid data vs invalid data threshold. It's the end of write to the log segment ratio.
For that segment, it's possible that all data are valid but we didn't write the full log segment and wasted space.